src/components/expenses/ExpenseSelect.tsx

```diff
@@ -0,0 +1,75 @@
+import { useTranslation } from 'react-i18next'
+import { FormControl, MenuItem, TextFieldProps } from '@mui/material'
+import { useField } from 'formik'
+
+import { useDocumentsList } from 'common/hooks/documents'
+import { useVaultsList } from 'common/hooks/vaults'
+import { usePersonList } from 'common/hooks/person'
+import { ExpenseCurrency, ExpenseStatus, ExpenseType } from 'gql/expenses'
+import FormTextField from 'components/common/form/FormTextField'
+
+type Props = {
+  name: string
+  allowEmpty: boolean
+} & TextFieldProps
+
+export default function ExpenseSelect({ name, allowEmpty, ...TextFieldProps }: Props) {
+  const { t } = useTranslation('expenses')
+
+  const validTypes = Object.keys(ExpenseType)
+  const validStatuses = Object.keys(ExpenseStatus)
+  const validCurrencies = Object.keys(ExpenseCurrency)
+
+  const values =
+    name == 'documentId'
+      ? useDocumentsList().data?.map((record) => record.id)
+      : name == 'vaultId'
+      ? useVaultsList().data?.map((record) => record.id)
+      : name == 'type'
+      ? validTypes
+      : name == 'status'
+      ? validStatuses
+      : name == 'currency'
+      ? validCurrencies
+      : ['true', 'false'] //deleted
```

Comment:
This part is not really well structured

Instead of having one `ExpenseSelect` component which combines `vaultId`, `type`, `status`, `currency` and etc we need to have separate components like:

- `VaultSelect`
- `ExpenseTypeSelect`
- `ExpenseStatusSelect`
- `CurrencySelect`
- `DeletedCheckbox`
- `DocumentSelect`
- `<PersonSelect name="approvedById" />`

This needs to be updated in order to accept the PR

Reply:
Is the `DeletedCheckbox` necessary on the form?

I don't receive the _soft_ deleted records because that's the way the API is set.

controller:
```javascript
@Public()
@Get('list')
async findAll() {
  return await this.expensesService.listExpenses()
}
```

service:
```javascript
async listExpenses(returnDeleted = false): Promise<Expense[]> {
  return this.prisma.expense.findMany({ where: { deleted: returnDeleted } })
}
```

I also cannot edit record which was _soft_ deleted because it just doesn't show on the grid.

Reply:
I've referenced that `DeleteCheckbox` as an example for the following piece of code in this branch

```tsx
<ExpenseSelect name="deleted" allowEmpty={false} disabled={true} />
```

Reply:
Beside that we don't need the `deleted` field to be visible

Reply:
I see.